crypto/evp/p_lib.c

```diff
@@ -1459,3 +1459,131 @@ int evp_pkey_downgrade(EVP_PKEY *pk)
     return 0;     /* No downgrade, but at least the key is restored */
 }
 #endif  /* FIPS_MODE */
+
+const OSSL_PARAM *EVP_PKEY_gettable_params(EVP_PKEY *pkey)
+{
+    if (pkey == NULL
+        || pkey->keymgmt == NULL
+        || pkey->keydata == NULL)
+        return 0;
+    return evp_keymgmt_gettable_params(pkey->keymgmt);
+}
+
+/*
+ * This value is set into param->return_size to indicate if
+ * evp_keymgmt_get_params() ignored the parameter.
+ * The EVP_PKEY_get_XXXX_param() methods return 0 if the parameter was ignored.
+ */
+#define TERMINATOR ((size_t)-1)
+
+int EVP_PKEY_get_bn_param(EVP_PKEY *pkey, const char *key_name, BIGNUM **bn)
+{
+    unsigned char buffer[1024];
```

Comment:
This shouldn't be a fixed size buffer.  I'd not be surprised if BN's greater than 8192 bits were required in the future.  I've certainly seen RSA keys long that this.

Reply:
Strongly agree.

Reply:
Doubled the size so it now supports 16K.
Also added a TODO - this is a general issue with passing BIGNUM's. All other types have their own data_size.